packages/api/lib/writeRecords/write-granules.js

```diff
@@ -226,9 +227,6 @@ const _removeExcessFiles = async ({
   knex,
   writtenFiles,
 }) => {
-  if (writtenFiles.length === 0) {
-    throw new Error('_removeExcessFiles called with no written files');
-  }
```

Comment:
I think this is right. This was removed on main and we don't want to throw an error because we'll just call deleteExcluding with an empty array. Just thinking out loud.